File: Python3/Find Common Characters.py
//1
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        check = list(words[0])
        for word in words[1:]:
            newCheck = []
            for c in word:
                if c in check:
                    newCheck.append(c)
                    check.remove(c)
            check = newCheck
        
        return check

//2
class Solution:
    def commonChars(self, words: List[str]) -> List[str]:
        if len(words) < 2:
            return words
        res = []
        word1 = set(words[0])
        for char in word1:
            frequency = min([word.count(char) for word in words])
            res += [char] * frequency
        return res

File: Golang/Find Common Characters.go
func commonChars(words []string) []string {
    var res []string
    
    arr := make([][26]byte, len(words))
    
    for i := 0; i < len(words); i++ {
        for j := 0; j < len(words[i]); j++ {
            arr[i][words[i][j] - 'a']++
        }
    }
    
    for i := 0; i < 26; i++ {
        isValid, count := true, 1<<63-1
        
        for j := 0; j < len(words); j++ {
            if arr[j][i] <= 0 {
                isValid = false
                break
            } else if int(arr[j][i]) < count {
                count = int(arr[j][i])
            }
        }
        
        if isValid {
            for count > 0 {
                res = append(res, fmt.Sprintf("%c", i + 'a'))
                count--
            }
        }
    }
    
    return res
}